monai/handlers/mlflow_handler.py

```diff
@@ -236,10 +242,20 @@ def start(self, engine: Engine) -> None:
     def _set_experiment(self):
         experiment = self.experiment
         if not experiment:
-            experiment = self.client.get_experiment_by_name(self.experiment_name)
-            if not experiment:
-                experiment_id = self.client.create_experiment(self.experiment_name)
-                experiment = self.client.get_experiment(experiment_id)
+            for _ in range(3):
+                try:
+                    experiment = self.client.get_experiment_by_name(self.experiment_name)
+                    if not experiment:
+                        experiment_id = self.client.create_experiment(self.experiment_name)
+                        experiment = self.client.get_experiment(experiment_id)
+                    break
+                except MlflowException as e:
+                    if "RESOURCE_ALREADY_EXISTS" in str(e):
+                        logger.warning("Experiment already exists; delaying before retrying.")
+                        time.sleep(1)
+                        continue
```

Comment:
I don't think this line is needed. Furthermore, if the exception is raised three times in a row the loop will silently do nothing. In place of this line we should check if this is the last time through the loop and raise the exception if so instead of sleeping. The check after this loop isn't sufficient since it raises a different exception which won't have the  information that the MlflowException would have.

Reply:
Thanks @ericspod! A good point, I removed the line and raise error if _retry_time meet 3. Could you please help review it again? Thanks.

Reply:
I think it's good now, thanks!